File: Level2/N개의 최소 공배수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;
int gcd(int a, int b);
int lcm(int a, int b);

int solution(vector<int> arr) {
    int answer = 0;
    
    //오름차순 정렬
    sort(arr.begin(), arr.end());
    answer = arr[0];
    
    for(int i = 1; i < arr.size(); i++)
    {
        answer = lcm(answer, arr[i]);
    }
    return answer;
}

int gcd(int a, int b)
{
    while(b != 0)
    {
        int r = a % b;
        a = b;
        b = r;
    }
    return a;
}

int lcm(int a, int b)
{
    return a * b / gcd(a, b);
}

File: Level2/N개의 최소 공배수.js
function gcd(a, b){
    let c = 0;
    
    while(b != 0)
    {
        c = a % b;
        a = b;
        b = c;
    }
    return a;
}

function lcm(a, b){
    return a * b / gcd(a, b);
}

function solution(arr) {
    let answer = 0;
    
    arr.sort((a, b) => a - b);
    
    for(let i = 0; i < arr.length - 1; i++){
        answer = lcm(arr[i], arr[i + 1]);
        arr[i + 1] = answer;
    }
    
    return answer;
}